teachertool/src/services/makecodeEditorService.ts

```diff
@@ -40,13 +40,13 @@ export const sendMessageAsync = (message?: any) => {
     });
 }
 
-export const getIframeRef = (ref: HTMLIFrameElement) => {
-    makecodeEditorRef = ref;
-    window.addEventListener("message", onMessageReceived);
-    makecodeEditorRef.addEventListener("load", () => {console.log("loaded")});
-}
-
-export const removeIframeRef = () => {
-    makecodeEditorRef = undefined;
-    window.removeEventListener("message", onMessageReceived);
+export const setEditorRef = (ref: HTMLIFrameElement | null) => {
+    if (ref) {
```

Comment:
If `makecodeEditorRef` is already set, it needs to be torn down. Listeners removed, etc.

Reply:
So, Thomas had a good point that the event listeners are on the window rather than the on the ref.. Would those listeners still need to be removed? 

Also, if we do this, it kinda sounds like we should just set `makeCodeEditorRef` to undefined each time we set the editor ref. As in, the first thing we do in the function is `makecodeEditorRef = undefined` since no matter if there's a ref passed in or not, we'll want the `makecodeEditorRef` to be clean.